File: result/quixbugs_result/python/breadth_first_search.py
def breadth_first_search(startnode, goalnode):
    queue = Queue()
    queue.append(startnode)

    nodesseen = set()
    nodesseen.add(startnode)

    while queue:
        node = queue.popleft()

        if node is goalnode:
            return True
        else:
            new_nodes = [n for n in node.successors if n not in nodesseen]
            queue.extend(new_nodes)
            nodesseen.update(new_nodes)

    return False

File: result/quixbugs_result/java/BREADTH_FIRST_SEARCH.java
public static boolean breadth_first_search(Node startnode, Node goalnode) {
    Deque<Node> queue = new ArrayDeque<>();
    queue.addLast(startnode);

    nodesvisited.add(startnode);

    while (!queue.isEmpty()) {
        Node node = queue.removeFirst();

        if (node == goalnode) {
            return true;
        } else {
            for (Node successor_node : node.getSuccessors()) {
                if (!nodesvisited.contains(successor_node)) {
                    queue.addFirst(successor_node);
                    nodesvisited.add(successor_node);
                }
            }
        }
    }
    return false;
}